rxjs/src/app/app.component.ts

```diff
@@ -0,0 +1,83 @@
+import { Component, OnDestroy, OnInit } from '@angular/core';
+import {
+  combineLatest,
+  filter,
+  forkJoin,
+  map,
+  mergeMap,
+  Observable,
+  Subject,
+  Subscription,
+  switchMap,
+  debounceTime
+} from 'rxjs';
+import { MockDataService } from './mock-data.service';
+
+@Component({
+  selector: 'my-app',
+  templateUrl: './app.component.html',
+  styleUrls: ['./app.component.css'],
+})
+export class AppComponent implements OnInit, OnDestroy {
+  searchTermByCharacters = new Subject<string>();
+  charactersResults$: Observable<any>;
+  planetAndCharactersResults$: Observable<any>;
+  loadingSubscription?: Subscription;
+  isLoading: boolean = false;
+
+  constructor(private mockDataService: MockDataService) {}
+
+  ngOnInit(): void {
+    this.initLoadingState();
+    this.initCharacterEvents();
+  }
+
+  changeCharactersInput(element): void {
+    const inputValue: string = element.target.value;
+    this.searchTermByCharacters.next(inputValue)
+  }
+
+  initCharacterEvents(): void {
+    this.charactersResults$ = this.searchTermByCharacters
+      .pipe(filter(searchText => searchText.length > 3), 
+        debounceTime(500),
+        mergeMap((searchText) => this.mockDataService.getCharacters(searchText))
```

Comment:
You should use other type of "Map" function. We want to drop the previous request every time we enter a new text into a search bar (we want to switch between them).